### PR TITLE
Bug 2091086: test/extended/cli/explain: Include OperatorHub CRD when 'marketplace' is enabled

### PR DESCRIPTION
**Pull request by wking**
Like 6d1143a0f2 (#26998), but catching up to openshift/api@d905557541 (openshift/api#1178) and operator-framework/operator-marketplace@eca17739ed (operator-framework/operator-marketplace#472) dancing the OperatorHub CRD over to the marketplace operator, and making it part of the `marketplace` cluster-version capability.  Plus an openshift/api vendor bump to bring in the `ClusterVersionCapabilityMarketplace` constant for that capability).

### CODE REVIEW

File: test/extended/cli/explain.go [hunk 1]
```diff
@@ -427,8 +430,12 @@ func getCrdTypes(oc *exutil.CLI) []schema.GroupVersionResource {
 	crdTypes = append(crdTypes, machineTypes...)
 	crdTypes = append(crdTypes, additionalOperatorTypes...)
 
-	// Only add metal3 types when the baremetal capability is enabled
+	// Conditional, capability-specific types
 	for _, capability := range clusterVersion.Status.Capabilities.EnabledCapabilities {
+		if capability == configv1.ClusterVersionCapabilityMarketplace {
+			crdTypes = append(crdTypes, marketplaceTypes...)
+			break
+		}
```
Comment (njhale) [inline test/extended/cli/explain.go:438]: nit: you could get fancy and use a switch statement for extra XP :)

Comment (wking) [reply test/extended/cli/explain.go]: > ...use a switch statement...

Done in abf3b8ea6f -> 160ce18f5a.

File: test/extended/cli/explain.go [hunk 2]
```diff
@@ -427,8 +430,12 @@ func getCrdTypes(oc *exutil.CLI) []schema.GroupVersionResource {
 	crdTypes = append(crdTypes, machineTypes...)
 	crdTypes = append(crdTypes, additionalOperatorTypes...)
 
-	// Only add metal3 types when the baremetal capability is enabled
+	// Conditional, capability-specific types
 	for _, capability := range clusterVersion.Status.Capabilities.EnabledCapabilities {
+		if capability == configv1.ClusterVersionCapabilityMarketplace {
+			crdTypes = append(crdTypes, marketplaceTypes...)
+			break
```
[the remark this answers is not in the document]
Comment (wking) [reply test/extended/cli/explain.go]: > ...we'll need to drop these `break` statements...

Done in abf3b8ea6f -> 160ce18f5a.

**Pull request merged**